File: Nordeus-Challenge/Assets/_Scripts/Models/Combat/EnvironmentEffect.cs
using System;
using Newtonsoft.Json;

[Serializable]
public class EnvironmentEffect
{
    [JsonProperty("resourceType")] public string resourceType; // "HEALTH" | "MANA" | "STAMINA"
    [JsonProperty("effectType")]   public string effectType;   // "GAIN" | "LOSE"
    [JsonProperty("amount")]       public int    amount;
    [JsonProperty("isPercent")]    public bool   isPercent;
}

File: backend/src/main/java/com/example/backend/models/EnvironmentEffect.java
package com.example.backend.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class EnvironmentEffect {

    public enum ResourceType { HEALTH, MANA, STAMINA }
    public enum EffectType   { GAIN, LOSE }

    private ResourceType resourceType;
    private EffectType   effectType;
    private int          amount;
    private boolean      isPercent = false;
}
